R/Utilities.R

```diff
@@ -646,3 +646,93 @@ getTADATemplate <- function(){
   colsreq = names(TADA::Nutrients_Utah)[!grepl("TADA.",names(TADA::Nutrients_Utah))]
   writexl::write_xlsx(TADA::Nutrients_Utah[1,colsreq], path = "TADATemplate.xlsx")
 }
+
+
+#' Identify and group nearby monitoring locations
+#'
+#' This function takes a TADA dataset and creates a distance matrix for all sites in the dataset. It then uses the buffer input value to determine which sites are within the buffer and should be identified as part of a nearby site group.
+#'
+#' @param .data TADA dataframe OR TADA sites dataframe
+#' @param dist_buffer Numeric. The maximum distance (in meters) two sites can be from one another to be considered "nearby" and grouped together.
+#'
+#' @return Input dataframe with one or more TADA.SiteGroup columns (depending upon if a site belongs to more than one nearby group), which identify each site group as "Group_#" for easy filtering. 
```

Comment:
discuss automatically updating TADA.MonitoringLocationIdentifier to the new MonitoringLocationIdentifier for the grouping, if user desires to assess the grouping as a "single" site